se_core/include/se/node_iterator.hpp

```diff
@@ -216,50 +237,84 @@ class node_iterator {
 
     return false;
   }
-  /**
-   * update all frontier voxels, targeting the frontier voxel just above floor
-   * @param blockCoord bottom left voxel block (node) voxel
-   * @return
-   */
-  bool deleteFrontierVoxelBlockviaMorton(const Eigen::Vector3i &blockCoord) {
+/**
+ * update all frontier voxels, targeting the frontier voxel just above floor
+ * @param blockCoord bottom left voxel block (node) voxel
+ * @return
+ */
+  bool deleteFrontierVoxelBlockviaMorton(const int64_t morton) {
 //    VoxelBlock<T>* block = map_.fetch(blockCoord(0), blockCoord(1), blockCoord(2));
     bool has_frontier = false;
 
-    int xlast = blockCoord(0) + BLOCK_SIDE;
-    int ylast = blockCoord(1) + BLOCK_SIDE;
-    int zlast = blockCoord(2) + BLOCK_SIDE;
-    se::VoxelBlock<T> *block = map_.fetch(blockCoord(0), blockCoord(1), blockCoord(2));
-    for (int z = blockCoord(2); z < zlast; ++z) {
-      for (int y = blockCoord(1); y < ylast; ++y) {
-        for (int x = blockCoord(0); x < xlast; ++x) {
-          // make handler with the current voxel
-          VoxelBlockHandler<T> handler = {block, Eigen::Vector3i(x, y, z)};
-          auto data = handler.get();
+    const Eigen::Vector3i blockCoord = keyops::decode(morton);
+    Node<T> *node = nullptr;
+    bool is_block = false;
+    // returned node could be at any level, but currently only frontier node at leaf level
+    map_.fetch_octant(blockCoord(0), blockCoord(1), blockCoord(2), node, is_block);
+
+    if (!is_block) {
```

Comment:
It would be easier to follow if the if and else were swapped so that the test is `if (is_block)`.